File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        // 정규표현식으로 영숫자만을 걸러낸다.
        String filtered_s = s.replaceAll("[^A-Za-z0-9]", "").toLowerCase();
        // 문자열을 뒤집는다.
        String revesed_filtered_s = new StringBuilder(filtered_s).reverse().toString();
        // 팰린드롬 검증
        return filtered_s.equals(revesed_filtered_s);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var start = 0
        var end = s.length -1
        
        // 서로 중앙으로 이동해 나가다 겹치는 지점에 도달하면 종료
        while(start < end) {
            when {
                // 영숫자인지 판별, 유효하지 않다면 한 칸 이동
                !Character.isLetterOrDigit(s[start]) -> {
                    start++
                }
                // 영숫자인지 판별, 유효하지 않다면 한 칸 이동
                !Character.isLetterOrDigit(s[end]) -> {
                    end--
                }
                else -> {
                    // 이 외에는 유효한문자열, 소문자로 변경하여 두문자가 같은지 검증한다.
                    if(Character.toLowerCase(s[start]) != Character.toLowerCase(s[end])) {
                        return false
                    }
                    
                    start++
                    end--
                }
            }
        }
        
        return true
    }
}